packages/components/tests/helpclasses/base-test-help.js

```diff
@@ -41,223 +41,278 @@ export default class BaseTestHelp {
     return new Options(UIOptionDefaults, ta)
   }
 
-  static baseVuexStore () {
-    return new Vuex.Store({
-      modules: {
-        popup: {
-          namespaced: true,
-          state: {
-            visible: false
+  /**
```

Comment:
I don't suppose there is any way to automatically generate the mock for the Vuex store from the code itself?

Reply:
I was following the practices currently established in the `BaseTestHelp` class, but I'm wondering if real Vuex objects, not mocks, would be nearly as performant. I probably can run a series of tests to see how would it be if we'll use real life objects instead of the mocks. If real life objects be simple, there will be no serious performance impact. Making them simple will benefit the clarity of the code as well. So if the code is too heavy and complex to be used in tests maybe we should simplify and split it until it becomes lightweight enough? What if we try to make real business object mockable (not sure we can but we can try, we'll need to make things that are not suitable for tests easily replaceable during test time as we did with the `z-index` method) instead of creating mocks for them?

Reply:
I think we need to also ask ourselves what really needs to be mocked and whether the fact that we have this need means we still have too tightly coupled code.